backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/banneralert/internal/BannerAlert.kt

```diff
@@ -0,0 +1,135 @@
+/*
+* Backpack for Android - Skyscanner's Design System
+*
+* Copyright 2018 Skyscanner Ltd
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*   http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+package net.skyscanner.backpack.compose.banneralert.internal
+
+import android.content.res.Configuration.UI_MODE_NIGHT_NO
+import android.content.res.Configuration.UI_MODE_NIGHT_YES
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.tooling.preview.PreviewParameter
+import androidx.compose.ui.tooling.preview.PreviewParameterProvider
+import net.skyscanner.backpack.compose.banneralert.BpkBannerAlertStyle
+import net.skyscanner.backpack.compose.banneralert.BpkBannerAlertType
+import net.skyscanner.backpack.compose.icon.BpkIcon
+import net.skyscanner.backpack.compose.text.BpkText
+import net.skyscanner.backpack.compose.theme.BpkTheme
+import net.skyscanner.backpack.compose.tokens.BpkBorderRadius
+import net.skyscanner.backpack.compose.tokens.BpkSpacing
+import net.skyscanner.backpack.compose.tokens.CloseCircle
+import net.skyscanner.backpack.compose.tokens.InformationCircle
+import net.skyscanner.backpack.compose.tokens.TickCircle
+
+@Composable
+internal fun BannerAlert(
+    type: BpkBannerAlertType,
+    message: String,
+    style: BpkBannerAlertStyle,
+    alertTypeContentDescription: String,
+    modifier: Modifier = Modifier,
+    icon: BpkIcon? = null,
+) {
+
+    val background = when (style) {
```

Comment:
actually these are not correct and also explain why the background wasn't working on the tests/docs:
there are some internal tokens - they are already generated in `BpkInfoBannerColors` - you need to use these, rather than the default background colours.
then you can also use the `canvasDefault` + `canvasContrast` background on the sample (and tests), like figma uses and it looks less jarring, especially on the demo

Reply:
Updated.